packages/kit/src/core/adapt/builder.js

```diff
@@ -51,6 +51,7 @@ export function create_builder({
 		/** @type {import('types').RouteDefinition} */
 		const facade = {
 			id: route.id,
+			type: route.endpoint ? 'endpoint' : 'page',
```

Comment:
it needs to be a pair of booleans, not an enum — a route can have both `+page` _and_ `+server` (in which case we do content negotiation)

Reply:
Design wise I think this should be an enum with three values. Two booleans is incorrect in the sense that both can never be false at the same time

Reply:
It's not future-proof — if we added a third route type (`socket` or something?), `both` would no longer make sense. Booleans are a better reflection of the types we use internally (where we went down this road before and switched to booleans because of the content negotiation thing)

Reply:
Tried again with booleans. Should the condition for `hasPage` be as simple as it is?

Also, is there a way around not false-positive-ing on a route with a page and an endpoint with a method other than `GET`?

Reply:
good point re false positives. I guess we would need to do something like

```
{
  pageMethods: ['GET', 'POST'],
  endpointMethods: ['DELETE']
}

Reply:
I like that solution. Let me investigate.

Reply:
Wait, is it possible for a page to have anything but `GET`?

Reply:
yes, if it has form actions

Reply:
Implemented and pushed. I went with a _slightly_ different object structure:

```ts
{
  endpoint: { methods: [] },
  page: { methods: [] }
}
```

Happy tochange it to the keys you suggested -- this just seemed more likely to grow without warts in the future.